packages/yoroi-extension/app/api/ada/lib/storage/bridge/delegationUtils.js

```diff
@@ -100,35 +99,35 @@ export function filterAddressesByStakingKey<T: { +address: string, ... }>(
 }
 
 export async function getUtxoDelegatedBalance(
-  publicDeriver: PublicDeriver<>,
+  wallet: WalletState,
   stakingAddress: string,
 ): Promise<MultiToken> {
-  const withUtxos = asGetAllUtxos(publicDeriver);
-  if (withUtxos == null) {
-    return publicDeriver.getParent().getDefaultMultiToken();
-  }
-  const basePubDeriver = withUtxos;
-
   // TODO: need to also deal with pointer address summing
   // can get most recent pointer from getCurrentDelegation result
 
-  const stakingKey = unwrapStakingKey(stakingAddress);
-  const allUtxo = await basePubDeriver.getAllUtxos();
+  const stakingKey = unwrapStakingKey(wallet.stakingAddress);
+  const allUtxo = wallet.utxos;
   const allUtxosForKey = filterAddressesByStakingKey<ElementOf<IGetAllUtxosResponse>>(
     stakingKey,
     allUtxo,
     false,
   );
+  const defaultToken = {
+    defaultNetworkId: wallet.networkId,
+    defaultIdentifier: wallet.defaultTokenId,
+  };
+  const defaultMultiToken = new MultiToken([], defaultToken);
```

Comment:
This particular pattern repeats really often. I'd really consider adding:

`wallet.defaultToken`

and 

`wallet.newDefaultMultiToken()`

Or something like that